addons/purchase/models/purchase_order_line.py

```diff
@@ -620,7 +572,7 @@ def _prepare_purchase_order_line(self, product_id, product_qty, product_uom, com
             'name': name,
             'product_qty': uom_po_qty,
             'product_id': product_id.id,
-            'product_uom_id': product_id.uom_po_id.id,
+            'product_uom_id': seller.product_uom_id.id or product_uom.id,
```

Comment:
if I understood your math earlier, i.e.
`uom_po_qty = product_uom._compute_quantity(product_qty, product_id.uom_id, rounding_method='HALF-UP')`
you converted `uom_po_qty` to `product_id.uom_id`, so you're returning the wrong `uom` here